aeron-cluster/src/main/java/io/aeron/cluster/Election.java

```diff
@@ -1121,6 +1170,33 @@ private void updateRecordingLog(final long nowNs)
         }
     }
 
+    private void updateRecordingLogForReplication(final long leadershipTermId, final long logPosition, final long nowNs)
+    {
+        final RecordingLog recordingLog = ctx.recordingLog();
+        final long timestamp = ctx.clusterClock().timeUnit().convert(nowNs, TimeUnit.NANOSECONDS);
```

Comment:
I wonder why do we need to convert a `nowNs` to `NANOSECONDS`? Isn't it a no op?

Reply:
The Cluster TimeUnit may not be nanoseconds, default is milliseconds, so we need to covert the nanos to whatever the cluster timestamp unit is.

Reply:
True, we are converting from the nanoseconds to the cluster time.